software/app/src/main/java/com/example/signbuddy/MainActivity.java

```diff
@@ -12,51 +12,79 @@
 import android.content.Intent;
 import android.content.pm.PackageManager;
 import android.os.Bundle;
+import android.os.Environment;
 import android.util.Log;
 import android.view.View;
 import android.widget.Button;
+import android.widget.EditText;
 import android.widget.ProgressBar;
 import android.widget.Toast;
 
 import com.clj.fastble.BleManager;
 import com.clj.fastble.callback.BleGattCallback;
 import com.clj.fastble.callback.BleNotifyCallback;
 import com.clj.fastble.callback.BleScanCallback;
+import com.clj.fastble.callback.BleWriteCallback;
 import com.clj.fastble.data.BleDevice;
 import com.clj.fastble.exception.BleException;
 import com.clj.fastble.scan.BleScanRuleConfig;
 
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.locks.ReentrantLock;
 
 public class MainActivity extends AppCompatActivity {
 
-    private Button connectButton;
-    private ProgressBar connectProgress;
     private final int FINE_LOCATION_CODE = 1;
-    private BleDevice SignBuddy;
     private final String ble_name = "Sign Buddy BLE";
     private final String uuid_service = "6E400001-B5A3-F393-E0A9-E50E24DCCA9E";
+    private final String uuid_characteristic_write = "6E400002-B5A3-F393-E0A9-E50E24DCCA9E";
     private final String uuid_characteristic_notify = "6E400003-B5A3-F393-E0A9-E50E24DCCA9E";
-    private int byteCount = 0;
+    private final byte[] cmd_sample_once = {0x01};
```

Comment:
capitalize these so we know they are constants

Reply:
Will do